src/arch/ia32e/interrupts/frame.rs

```diff
@@ -0,0 +1,42 @@
+use crate::arch::ia32::address::VirtualAddress;
+use core::fmt;
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct InterruptStackFrameValue {
+    /// Next or faulting instruction
+    pub rip: VirtualAddress,
+    /// Code segment selector (CS)
+    pub code_segment: u64,
+    /// CPU flags
+    pub rflags: u64,
+    /// Stack pointer (RSP)
+    pub stack_pointer: VirtualAddress,
+    /// Stack segment (SS)
+    pub stack_segment: u64,
+}
+
+impl fmt::Display for InterruptStackFrameValue {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "RIP: {}\nCS: {:X}\nRLAGS: {:X}\nRSP: {}\nSS: {:X}\n",
+            self.rip,
+            self.code_segment,
+            self.rflags,
+            self.stack_pointer,
+            self.stack_segment
+        )
+    }
+}
+
+#[repr(C)]
+pub struct InterruptStackFrame {
```

Comment:
I'm not sure why we need this intermediate structure ? Couldn't we simply rename `InterruptStackFrameValue` to `InterruptStackFrame` and remove this structure entirely ?

Reply:
Yes, sorry, I should have tried. Fixed in a31d94703ddd20bded81afb35a08691a93636c5b